### DISCUSSION TITLE
Setup Error: No Module Named Graphite or django-admin.py missing

### DISCUSSION THREAD

**Discussion by dishorned** [graphite-project/graphite-web#2765] [Q&A]
Hey Guys,
I hope you are all doing well.

I want to set up a new, clean Debian VM and only install Graphite on it. I tried a lot of different szenarios and tutorials including installing it with pip or from source (cannot install it with the apt package manager, because graphite-web is not in the debian 11 repository), but nothing works.

But let's go a step back. I am using a fresh installed debian 11 VM, which is up to date, with no special software installed. For getting graphite up and running, I installed all the required dependencies showing up in the check-dependencies.py and additional to it, apache2.

After all dependencies are met, I installed carbon with requirements (including whisper) and graphite-web:

`git clone https://github.com/graphite-project/carbon`
`git clone https://github.com/graphite-project/graphite-web`
`cd carbon`
`pip install -r requirements.txt`
`python3 setup.py install`
`cd ..`
`cd graphite-web`
`python3 setup.py install`

At this point, everything seems good. I want to use a mariadb instead of sqllite, so I changed that in the local_settings.py (/opt/graphite/webapp/graphite), which I copied from template. I also changed the secret-key and timezone.

Now the problem starts. Different tutorials show different methods to get the db up and running. One tutorial says, to set it up using the graphite-manage command, but my system cannot find the command. So I also tried the more common way from the official tutorial:

`PYTHONPATH=$GRAPHITE_ROOT/webapp django-admin.py migrate --settings=graphite.settings`

Here it says, that the django.admin.py command is not found, but python3-django is installed. Does something is missing?

I also tried another way to set it up with the manage.py script, but that is showing the same error, when I try to directly run the settings.py script. It always says ModuleNotFoundError: No module named graphite.

I also tried it with python 2.7, because that is recommanded in the requirements.txt, but with no luck. I don't want to use docker, because I am not a linux pro and first I want to learn a lot about the system itself and how to manually (from source etc) install software, dependencies and solve errors.

Help is very appreciated and I hope you guys now, what I am doing wrong or what is missing.

Thanks a lot and best regards

**Comment by bucko909** [comment]
Our install instructions suggest using `/path/to/python-virtualenv-with-graphite-installed/bin/python manage.py migrate` from within the `webapp` directory. This does complete and create the sqlite database, so I assume it'd work in your configuration, too, but I can't guarantee this works 100%, as we don't use tags at all on our instance, so the `sqlite` database is only created to allow the app to start up.

**Reply by dishorned** [reply]
Thanks a lot for your answer. As in the meanwhile, Debian 12 has been released, I've tried the installation form apt repository  and it worked perfectly.
Maybe this is not the recommanded way where I can stay uptodate with the latest release, but for the moment it's fine. When I someday reinstall it, I may try a different approach.

Thanks and best regards
